stdlib/test/collections/test_list.mojo

```diff
@@ -286,6 +286,60 @@ def test_list_reverse_move_count():
     # Keep vec alive until after we've done the last `vec.data + N` read.
     _ = vec^
 
+def test_list_index():
+    var test_list_a = List[Int](10,20,30,40,50)
+    
+    # Basic Functionality Tests
+    assert_equal(test_list_a.index(10), 0)  # Verify finding the first element's index.
+    assert_equal(test_list_a.index(30), 2)  # Verify finding the index of a middle element.
+    assert_equal(test_list_a.index(50), 4)  # Verify finding the last element's index.
+    with assert_raises():  # Verify error for a non-existent element.
+        _ = test_list_a.index(60)
+
+    # Tests With Start Parameter
+    assert_equal(test_list_a.index(30, start=1), 2)  # Verify search beginning from a specific start index.
+    assert_equal(test_list_a.index(30, start=-4), 2)  # Verify element search with a negative start index.
```

Comment:
**Question** Is this one correct?  I think in python,

```python
xs = [10, 20, 30, 40, 50]
xs.index(30, -4, 2)
```

will raise.

Reply:
Um, It seems you misread the test case. We are not calling `index(30, -4, 2)`. Instead, the method is `index(30, -4)`, and then we use `assert_equal` to verify whether the result equals `2`. So the final code for that test is correct as below:

```mojo
assert_equal(test_list_a.index(30, start=-4), 2)
```

Also if you run your given code in Mojo, it will raise an error just like Python. I believe that the implemented `index` method does not have any issues here.

Reply:
Oh you're definitely right here. Sorry I misread the test case.